app/controllers/comments_controller.rb

```diff
@@ -63,6 +65,13 @@ def set_comment
       @comment = Comment.find(params[:id])
     end
 
+    def is_an_authorized_user
+      @photo = Photo.find(params.fetch(:comment).fetch(:photo_id))
+      if current_user != @photo.owner && @photo.owner.private? && !current_user.leaders.include?(@photo.owner)
+        redirect_back fallback_location: root_url, alert: "Not authorized"
+      end
+    end
+    
     # Only allow a list of trusted parameters through.
```

Comment:
You have :author_id in comment_params but also explicitly assign @comment.author = current_user in the create action. This redundancy could lead to confusion or security issues if the author_id parameter is manipulated. It's safer to remove :author_id from comment_params since you're setting the author explicitly.